plugins/cache-ehcache/src/main/java/org/opensearch/cache/store/disk/EhcacheDiskCache.java

```diff
@@ -89,14 +94,14 @@ public class EhcacheDiskCache<K, V> implements ICache<K, V> {
     // A Cache manager can create many caches.
     private final PersistentCacheManager cacheManager;
 
-    // Disk cache
-    private Cache<K, V> cache;
+    // Disk cache. Because of a bug in ehcache, we have to store ByteArrayWrapper rather than byte[].
```

Comment:
Lets not call it a bug yet. Lets say "Using ByteArrayWrapper to compare two byte[] by values rather than reference checks"